Release version 3.1.0 (logistics)

lecoqlibre opened the pull request:
Add logistic concepts:
- `LocalizedProduct`.
- `AsPlannedLocalTransformation`, `AsPlannedLocalConsumptionFlow`, `AsPlannedLocalProductionFlow`.
- `PhysicalProduct`.
- `AsRealizedTransformation`, `AsRealizedConsumptionFlow`, `AsRealizedProductionFlow`.
- `TheoreticalStock`.
- `RealStock`.
- `ProductBatch`.
- `VirtualPlace`.
- `PhysicalPlace`.
- `PaymentMethod`.
- `PickupOption`.
- `DeliveryOption`.

lecoqlibre committed:
Start logistics

lecoqlibre committed:
Continue logistics

lecoqlibre committed:
Add delivery options and product loops interfaces

lecoqlibre committed:
Finish LocalizedProduct, add PhysicalProduct, add associated loops, add stocks

lecoqlibre committed:
Add ProductBatch and Places classes

lecoqlibre committed:
Add ShippingOptions and PaymentMethod

lecoqlibre committed:
Update changelog and supported

lecoqlibre committed:
Update supported

lecoqlibre committed:
Upadate changelog and supported

lecoqlibre committed:
Add begin and end dates to shipping options

RaggedStaff commented on SUPPORTED.md:
```diff
@@ -11,78 +14,130 @@ Here is a list of the DFC features that the model currently supports.
   - [x] longitude
   - [x] region
   - [ ] addressOf
-- `Agent`
```
Can you clarify why `Agent` has been removed from SUPPORTED.md  ? 😕

lecoqlibre replied:
I removed abstract classes and moved their properties to concrete classes for more clarity. It concerns `Agent`, `ShippingOption`, `DefinedProduct`.

lecoqlibre replied:
@RaggedStaff I only removed the Agent class from the Supported.md file, not from the UML model.

RaggedStaff reviewed (commented):
Thanks @lecoqlibre 

Can you clarify why the Agent items have been removed from the SUPPORTED.md file ?

Wondering if there's a way to auto-generate the Supported list, but I suspect not as you'd have to parse the ontology at that point, and then we could automate creation of the data model! 🙄

lecoqlibre commented:
> Wondering if there's a way to auto-generate the Supported list, but I suspect not as you'd have to parse the ontology at that point, and then we could automate creation of the data model! 🙄

Yes we could generate that file automatically. But it means we have to parse the ontology and this can be a huge work. I wrote it manually because I did not find any existing tool and time was limited.

In the future I would like to use SHACL shapes instead of UML. I think it would be easier to generate that file from there.

lecoqlibre committed:
Fix semantic stereotypes (#26)

Added to:
- ILocalizedProduct
- IPhysicalProduct
- IStock
- IProductBatch
- IPlace
- IShippingOption
- IPaymentMethod

lecoqlibre committed:
Update changelog (#27)
